flow/FileTraceLogWriter.cpp

```diff
@@ -87,7 +87,18 @@ void FileTraceLogWriter::write(const std::string& str) {
 void FileTraceLogWriter::open() {
 	cleanupTraceFiles();
 
-	auto finalname = format("%s.%d.%s", basename.c_str(), ++index, extension.c_str());
+	++index;
+	int indexWidth = -1;
+	while (index > 0) {
+		index /= 10;
+		++indexWidth;
+	}
```

Comment:
If I'm understanding this correctly, it's going to reset the index to 0 every time we roll the trace file. I think we shouldn't modify index inside the loop (and possibly we could use an explicit computation of the number of digits, something like `floor(log10(index)+1)` with the assertion that `index > 0`).

Reply:
Good catch. I wonder why this was looking like it works when I ran tests... Anyways, I think I fixed it.